### PR TITLE
Execute Indirect state - Phase 1

### PR DESCRIPTION
**Pull request by HansKristian-Work**

### CODE REVIEW

File: libs/vkd3d/vkd3d_private.h [hunk 1]
```diff
@@ -2099,12 +2099,20 @@ struct d3d12_command_list
...
     bool has_replaced_shaders;
-    bool has_valid_index_buffer;
+
+    struct
+    {
+        VkBuffer buffer;
+        VkDeviceSize offset;
+        DXGI_FORMAT dxgi_format;
+        VkIndexType vk_type;
+        bool is_non_null;
```
Comment (doitsujin) [inline libs/vkd3d/vkd3d_private.h:2109]: couldn't we use the VkBuffer being `VK_NULL_HANLDE` for this case instead? This seems somewhat redundant now.

File: libs/vkd3d/command.c [hunk 2]
```diff
@@ -12442,37 +12446,136 @@ CONST_VTBL struct ID3D12CommandSignatureVtbl d3d12_command_signature_vtbl =
...
+    }
+
+    if (requires_root_signature && !root_signature)
+    {
+        ERR("Command signature requires root signature, but is not provided.\n");
+        return E_INVALIDARG;
+    }
+    else if (!requires_root_signature && root_signature)
+    {
+        ERR("Command signature requires root signature, root signature must be NULL.\n");
```
Comment (doitsujin) [inline libs/vkd3d/command.c:12570]: "does not require root signature"?

**Review by doitsujin — CHANGES_REQUESTED** [review]
Small nits.

File: libs/vkd3d/vkd3d_private.h (hunk 1 above)
Comment (HansKristian-Work) [reply libs/vkd3d/vkd3d_private.h]: Fixed.

File: libs/vkd3d/command.c (hunk 2 above)
Comment (HansKristian-Work) [reply libs/vkd3d/command.c]: Fixed.

**Review by doitsujin — APPROVED** [review]

**Pull request merged**